askama_parser/src/lib.rs

```diff
@@ -131,6 +129,55 @@ impl fmt::Display for ParseError {
     }
 }
 
+/// This type is used to handle `nom` errors and in particular to add custom error messages.
+/// It used to generate `ParserError`.
+///
+/// It cannot be used to replace `ParseError` because it expects a generic, which would make
+/// `askama`'s users experience less good (since this generic is only needed for `nom`).
+#[derive(Debug)]
+pub(crate) struct ParsingError<I> {
```

Comment:
I'd like this commit squashed into the first commit (that introduces it), and I still think it should be renamed? I'm thinking maybe `ErrorContext` as I think this actually represents Askama-specific context that's spliced into nom errors in order to propagate some extra state across nom parser combinators.

Reply:
`ErrorContext` sounds good to me.